exporter/exporterhelper/queued_retry.go

```diff
@@ -266,8 +266,8 @@ func NewDefaultRetrySettings() RetrySettings {
 	}
 }
 
-func createSampledLogger(logger *zap.Logger) *zap.Logger {
-	if logger.Core().Enabled(zapcore.DebugLevel) {
+func createSampledLogger(logger *zap.Logger, lCfg SampledLoggerSettings) *zap.Logger {
+	if !lCfg.Enabled || logger.Core().Enabled(zapcore.DebugLevel) {
```

Comment:
I feel like there is a bit of contradictory behavior, it makes sense in terms of backwards compatibility and the code, but conceptually I am not sure (we might need to document it at least):

"Sampler is enabled and logs are in debug mode"

In this case the code will not enable the sampler, but since we have a very explicit configuration which literally says "I want sampling", not sure if this combination might confuse people

Reply:
I would at least document this behavior in case we will not be proposing the behavior change which will allow the property to overwrite debug mode setting